third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -810,10 +879,23 @@ void Pingponger::getDotPingponged() {
   // tightly scheduling the latencies.
 
   int64_t numOfDotLikeOps = scaledDotOps.size() + dotOps.size();
-  if (numOfDotLikeOps != 1) {
-    LDBG("Only handle a single of either dot or dot_scaled op");
+
+  if (numOfDotLikeOps < 1 || numOfDotLikeOps > 2) {
+    LDBG("Only handle one or two dotlike ops");
     return;
   }
+
+  if (numOfDotLikeOps == 2 && numStages == 4) {
+    if (transformChainedDotSchedule(builder, loc).failed()) {
+      LDBG("Encountered failure when trying the ChainedDot ping pong "
+           "cluster transformation");
+      return;
+    }
+    addAsymmetricSyncToLoop(builder, loc);
+  } else if (numOfDotLikeOps == 2) {
+    return;
```

Comment:
We can merge this with the above and do early return inside with `numStages != 4`.